src/commissioner/commissioner.cpp

```diff
@@ -102,22 +103,29 @@ static int DummyKeyExport(void *               aContext,
 Commissioner::Commissioner(const uint8_t *aPskcBin, int aKeepAliveRate)
     : mDtlsInitDone(false)
     , mRelayReceiveHandler(OT_URI_PATH_RELAY_RX, Commissioner::HandleRelayReceive, this)
+    , mUdpRxHandler(OT_URI_PATH_UDP_RX, Commissioner::HandleUdpRx, this)
     , mPetitionRetryCount(0)
     , mJoinerSession(NULL)
     , mKeepAliveRate(aKeepAliveRate)
+    , mUdpListenFd(-1)
 {
     sockaddr_in addr;
 
     memcpy(mPskcBin, aPskcBin, sizeof(mPskcBin));
     addr.sin_family      = AF_INET;
     addr.sin_addr.s_addr = htonl(INADDR_LOOPBACK);
-    addr.sin_port        = htons(kPortJoinerSession);
+    addr.sin_port        = htons(kJoinerSessionPort);
     mCoapAgent           = Coap::Agent::Create(SendCoap, this);
     mCoapToken           = rand();
     mCoapAgent->AddResource(mRelayReceiveHandler);
+    mCoapAgent->AddResource(mUdpRxHandler);
     mCommissionState = kStateInvalid;
     VerifyOrExit((mJoinerSessionClientFd = socket(AF_INET, SOCK_DGRAM, 0)) > 0);
     SuccessOrExit(connect(mJoinerSessionClientFd, reinterpret_cast<struct sockaddr *>(&addr), sizeof(addr)));
+    if (SetupProxyServer() != 0)
```

Comment:
We don't throw exceptions, so we should make the constructor always success. I suggest do such initialization in a separate method.